docs/painless/painless-guide/painless-datetime.asciidoc

```diff
@@ -0,0 +1,295 @@
+<<painless-operators-numeric, numeric operators>> to modify a long
+representation of a datetime. Use <<method-call-operator, method calls>> to
+modify a ZonedDateTime representation of a datetime. Note most modification
+methods for a ZonedDateTime return a new instance that needs assignment or
+immediate use.
+
+===== Datetime Modification Examples
+
+* Subtract three seconds from a long (milliseconds)
++
+[source,Painless]
+----
+long milliSinceEpoch = 434931330000L;
+milliSinceEpoch = milliSinceEpoch - 1000L*3L;
+----
++
+* Add three days to a datetime
++
+[source,Painless]
+----
+ZonedDateTime zdt =
+        ZonedDateTime.of(1983, 10, 13, 22, 15, 30, 0, ZoneId.of('Z'));
+ZonedDateTime updatedZdt = zdt.plusDays(3);
+----
++
+* Subtract 125 minutes from a datetime
++
+[source,Painless]
+----
+ZonedDateTime zdt =
+        ZonedDateTime.of(1983, 10, 13, 22, 15, 30, 0, ZoneId.of('Z'));
+ZonedDateTime updatedZdt = zdt.minusMinutes(125);
+----
++
+* Set the year on a datetime
++
+[source,Painless]
+----
+ZonedDateTime zdt =
+        ZonedDateTime.of(1983, 10, 13, 22, 15, 30, 0, ZoneId.of('Z'));
+ZonedDateTime updatedZdt = zdt.withYear(1976);
+----
+
+==== Elapsed Time
+
+A script's objectives may require the elapsed time (difference) between two
+datetimes. Use either two longs or two ZonedDateTimes to calculate an elapsed
+time. Use <<subtraction-operator, subtraction>> to calculate an elapsed time
+between two longs. Use <<painless-api-reference-shared-ChronoUnit, ChronoUnit>>
+to calculate the difference between two ZonedDateTimes.
+
+===== Elapsed Time Examples
+
+* Elapsed time for two longs (milliseconds)
++
+[source,Painless]
+----
+long startTimestamp = 434931327000L;
+long endTimestamp = 434931330000L;
+long differenceInMillis = endTimestamp - startTimestamp;
+----
++
+* Elapsed time in milliseconds for two datetimes
++
+[source,Painless]
+----
+ZonedDateTime zdt1 =
+        ZonedDateTime.of(1983, 10, 13, 22, 15, 30, 11000000, ZoneId.of('Z'));
+ZonedDateTime zdt2 =
+        ZonedDateTime.of(1983, 10, 13, 22, 15, 35, 0, ZoneId.of('Z'));
+long differenceInMillis = ChronoUnit.MILLIS.between(zdt1, zdt2);
+----
++
+* Elapsed time in days for two datetimes
++
+[source,Painless]
+----
+ZonedDateTime zdt1 =
+        ZonedDateTime.of(1983, 10, 13, 22, 15, 30, 11000000, ZoneId.of('Z'));
+ZonedDateTime zdt2 =
+        ZonedDateTime.of(1983, 10, 17, 22, 15, 35, 0, ZoneId.of('Z'));
+long differenceInDays = ChronoUnit.DAYS.between(zdt1, zdt2);
+----
+
+==== Datetime Comparison
+
+Comparisons of datetimes are often required to meet a script's objectives. Use
+either two longs or two ZonedDateTimes to do a datetime comparison. Use
+standard <<painless-operators-boolean, comparison operators>> to compare two
+long datetime representations. Use <<method-call-operator, method calls>> to
+compare two ZonedDateTime datetime representations.
+
+===== Datetime Comparison Examples
+
+* Comparison of two longs (milliseconds)
```

Comment:
Again, this seems too trivial to note as an example and not actually useful

Reply:
I'd like to leave this for the same reasons as stated two comments above.